connector/exceptionsconnector/connector_logs.go

```diff
@@ -0,0 +1,137 @@
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+)
+
+type logsConnector struct {
+	logsConsumer consumer.Logs
+	component.StartFunc
+	component.ShutdownFunc
+
+	logger *zap.Logger
+	ld     plog.Logs
+}
+
+func newLogsConnector(logger *zap.Logger) (*logsConnector, error) {
+	logger.Info("Building logs exceptionsconnector")
+
+	return &logsConnector{
+		logger: logger,
+		ld:     plog.NewLogs(),
+	}, nil
+}
+
+// Capabilities implements the consumer interface.
+func (c *logsConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// ConsumeTraces implements the consumer.Traces interface.
+// It aggregates the trace data to generate logs.
+func (c *logsConnector) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
+	c.logger.Debug("Consume traces")
+	sl := c.newScopeLogs()
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
+		rspans := traces.ResourceSpans().At(i)
+		resourceAttr := rspans.Resource().Attributes()
+		serviceAttr, ok := resourceAttr.Get(conventions.AttributeServiceName)
+		if !ok {
+			continue
+		}
+		serviceName := serviceAttr.Str()
+		ilsSlice := rspans.ScopeSpans()
+		for j := 0; j < ilsSlice.Len(); j++ {
+			ils := ilsSlice.At(j)
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				for l := 0; l < span.Events().Len(); l++ {
+					event := span.Events().At(l)
+					if event.Name() == "exception" {
+						c.attrToLogRecord(sl, serviceName, span.Attributes(), event.Attributes(), event.Timestamp())
+					}
+				}
+			}
+		}
+	}
+	c.exportLogs(ctx)
+	return nil
+}
+
+func (c *logsConnector) exportLogs(ctx context.Context) error {
+	c.logger.Debug("Exporting logs")
+	if err := c.logsConsumer.ConsumeLogs(ctx, c.ld); err != nil {
+		c.logger.Error("Failed ConsumeLogs", zap.Error(err))
+		return err
+	}
+	return nil
+}
+
+func (c *logsConnector) newScopeLogs() plog.ScopeLogs {
+	rl := c.ld.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	sl.Scope().SetName("exceptionsconnector")
+	return sl
+}
+
+func (c *logsConnector) attrToLogRecord(sl plog.ScopeLogs, serviceName string, spanAttr, eventAttr pcommon.Map, eventTs pcommon.Timestamp) plog.LogRecord {
+	logRecord := sl.LogRecords().AppendEmpty()
+
+	logRecord.SetTimestamp(eventTs)
+	logRecord.SetSeverityNumber(plog.SeverityNumberError)
+	logRecord.SetSeverityText("ERROR")
+
+	logRecord.Attributes().PutStr("service.name", serviceName)
+	logRecord.Attributes().PutStr("exception.stacktrace", getValue(eventAttr, "exception.stacktrace"))
+	logRecord.Attributes().PutStr("exception.type", getValue(eventAttr, "exception.type"))
+	logRecord.Attributes().PutStr("exception.message", getValue(eventAttr, "exception.message"))
+
+	for k, v := range extractHTTP(spanAttr) {
```

Comment:
Why not other transports as well, like gRPC?

Reply:
Doing it on a follow-up PR sounds good.

Reply:
Once you create an issue tracking this, mention it here and please mark this as resolved.